cmake/modules/hunter_add_repository.cmake

```diff
@@ -0,0 +1,81 @@
+include(CMakeParseArguments) # cmake_parse_arguments
+include(hunter_internal_error)
+include(hunter_fatal_error)
+include(hunter_test_string_not_empty)
+include(hunter_status_debug)
+include(hunter_status_print)
+include(hunter_install_repository)
+
+function(hunter_add_repository)
+  if(PROJECT_NAME OR NOT HUNTER_GATE_ROOT)
```

Comment:
With this check it's not possible to include subproject by `add_subdirectory` if both parent and child projects have `hunter_add_repository`. In this scenario subproject settings should be ignored since parent can trigger packages installation to root.